apps/prairielearn/src/ee/lib/billing/stripe.ts

```diff
@@ -0,0 +1,40 @@
+import Stripe from 'stripe';
+import { loadSqlEquiv, queryAsync } from '@prairielearn/postgres';
+
+import { config } from '../../../lib/config';
+import { selectUserById } from '../../../models/user';
+
+const sql = loadSqlEquiv(__filename);
+
+export function getStripeClient() {
+  if (!config.stripeSecretKey) {
+    throw new Error('Stripe is not configured.');
+  }
+
+  return new Stripe(config.stripeSecretKey, { apiVersion: '2022-11-15' });
+}
+
+export async function getOrCreateStripeCustomerId(
```

Comment:
There's a potential race condition in this function: if a user initiates two payments at almost exactly the same time, we could end up creating two customers for them. I was hesitant to add a lock to solve this, as we'd have to hold that lock while performing a network operation, which is something we generally want to avoid.

In practice, this is unlikely to happen, and if it does, it's not a huge deal: we'll just have accidentally split payments between two customers, which won't impact much besides making support activities in the Stripe console just slightly more annoying.

Reply:
Done in 217c5a6.